sdks/java/extensions/sql/src/test/java/org/apache/beam/sdk/extensions/sql/meta/provider/bigquery/BigQueryWriteIT.java

```diff
@@ -55,8 +65,89 @@
           .addNullableField("arr", FieldType.array(STRING))
           .build();
 
+  private static final Schema SOURCE_SCHEMA_TWO =
+      Schema.builder()
+          .addNullableField("c_bigint", INT64)
+          .addNullableField("c_tinyint", BYTE)
+          .addNullableField("c_smallint", INT16)
+          .addNullableField("c_integer", INT32)
+          .addNullableField("c_float", FLOAT)
+          .addNullableField("c_double", DOUBLE)
+          .addNullableField("c_decimal", DECIMAL)
+          .addNullableField("c_boolean", BOOLEAN)
+          .addNullableField("c_timestamp", FieldType.DATETIME.withMetadata("TS"))
+          .addNullableField("c_varchar", STRING)
+          .addNullableField("c_char", STRING)
+          .addNullableField("c_arr", FieldType.array(STRING))
+          .build();
+
   @Rule public transient TestPipeline pipeline = TestPipeline.create();
   @Rule public transient TestBigQuery bigQuery = TestBigQuery.create(SOURCE_SCHEMA);
+  @Rule public transient TestBigQuery bigQueryTestingTypes = TestBigQuery.create(SOURCE_SCHEMA_TWO);
+
+  @Test
+  public void testSQLTypes() {
+    BeamSqlEnv sqlEnv = BeamSqlEnv.inMemory(new BigQueryTableProvider());
+
+    String createTableStatement =
+        "CREATE TABLE TEST( \n"
+            + "   c_bigint BIGINT, \n"
+            + "   c_tinyint TINYINT, \n"
+            + "   c_smallint SMALLINT, \n"
+            + "   c_integer INTEGER, \n"
+            + "   c_float FLOAT, \n"
+            + "   c_double DOUBLE, \n"
+            + "   c_decimal DECIMAL, \n"
+            + "   c_boolean BOOLEAN, \n"
+            + "   c_timestamp TIMESTAMP, \n"
+            + "   c_varchar VARCHAR, \n "
+            + "   c_char CHAR, \n"
+            + "   c_arr ARRAY<VARCHAR> \n"
+            + ") \n"
+            + "TYPE 'bigquery' \n"
+            + "LOCATION '"
+            + bigQueryTestingTypes.tableSpec()
+            + "'";
+    sqlEnv.executeDdl(createTableStatement);
+    System.out.println(bigQueryTestingTypes.tableSpec());
+
+    String insertStatement =
+        "INSERT INTO TEST VALUES ("
+            + "9223372036854775807, "
+            + "127, "
+            + "32767, "
+            + "2147483647, "
+            + "1.0, "
+            + "1.0, "
+            + "123.45, "
+            + "TRUE, "
+            + "TIMESTAMP '2018-05-28 20:17:40', "
```

Comment:
Should probably test to microseconds here (`.123456`) as I hear there is some ambiguity in how they are handled.

Reply:
Gotcha, I know what you are concern about. Seems like BigQuery internally supports time at microsecond level.

I can only test time at millisecond level (`.123`) because that's the smallest unit that Joda library supports.